src/Engine/ProtoCore/Utils/ArrayUtils.cs

```diff
@@ -117,25 +118,43 @@ public static ClassNode GetGreatestCommonSubclassForArray(StackValue array, Runt
             return runtimeCore.DSExecutable.classTable.ClassNodes[orderedTypes.First()];
         }
 
-        public static Dictionary<int, StackValue> GetTypeExamplesForLayer(StackValue array, RuntimeCore runtimeCore)
+        public static IEnumerable<StackValue> GetTypeExamplesForLayer(StackValue array, RuntimeCore runtimeCore)
         {
-            Dictionary<int, StackValue> usageFreq = new Dictionary<int, StackValue>();
+            var usageFreq = new Dictionary<int, List<StackValue>>();
 
             if (!array.IsArray)
             {
-                usageFreq.Add(array.metaData.type, array);
-                return usageFreq;
+                usageFreq.Add(array.metaData.type, new List<StackValue> {array});
+
+                // return flattened list of unique values
+                return usageFreq.Values.SelectMany(x => x.Select(y => y));
```

Comment:
doing `return usageFreq.Values.SelectMany(x => x); won't produce the same result?

Reply:
I guess you're right.